scripts/test_hardware/test_integration.py

```
@@ -0,0 +1,235 @@
+from .protocol import (
+    Mode, request_status, parse_status, StatusResponse,
+)
+from .result import TestResult, TestReport, Verdict
+
+
+REQUEST_TIMEOUT = 5
+STATUS_TIMEOUT = 0.5
+
+
+def _get(url: str) -> tuple[int, str]:
+    req = urllib.request.Request(url)
```

Ruff (0.15.2)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)